indigo/indigo/src/main/scala/indigo/shared/BoundaryLocator.scala

```diff
@@ -80,42 +85,40 @@ final class BoundaryLocator(
         None
 
       case s: Sprite[_] =>
-        spriteBounds(s).map(rect => BoundaryLocator.findBounds(s, rect.position, rect.size, s.ref))
+        spriteBounds(s)
 
       case t: Text[_] =>
-        Option(textBounds(t)).map { rect =>
-
-          val offset: Int =
-            t.alignment match {
-              case TextAlignment.Left   => 0
-              case TextAlignment.Center => rect.size.width / 2
-              case TextAlignment.Right  => rect.size.width
-            }
-
-          BoundaryLocator.findBounds(t, rect.position, rect.size, t.ref + Point(offset, 0))
-        }
+        Option(textBounds(t))
 
       case _ =>
         None
     }
 
-  def groupBounds(group: Group): Rectangle =
-    group.children match {
-      case Nil =>
-        Rectangle.zero
-
-      case x :: xs =>
-        xs.foldLeft(findBounds(x)) { (acc, node) =>
-          (acc, findBounds(node)) match
-            case (Some(a), Some(b)) => Option(Rectangle.expandToInclude(a, b))
-            case (r @ Some(_), _)   => r
-            case (_, r @ Some(_))   => r
-            case (r, _)             => r
-        }.map(_.moveBy(group.position))
-          .getOrElse(Rectangle.zero)
-    }
+  /** Finds the bounds or returns a `Rectangle` of size zero for convenience.
+    */
+  def bounds(sceneNode: SceneNode): Rectangle =
+    findBounds(sceneNode).getOrElse(Rectangle.zero)
+
+  private def groupBounds(group: Group): Rectangle =
+    val rect =
+      group.children match {
+        case Nil =>
+          Rectangle.zero
+
+        case x :: xs =>
+          xs.foldLeft(findBounds(x)) { (acc, node) =>
+            (acc, findBounds(node)) match
+              case (Some(a), Some(b)) => Option(Rectangle.expandToInclude(a, b))
+              case (r @ Some(_), _)   => r
+              case (_, r @ Some(_))   => r
+              case (r, _)             => r
+          }.map(_.moveBy(group.position))
+            .getOrElse(Rectangle.zero)
+      }
+
+    BoundaryLocator.findBounds(group, rect.position, rect.size, group.ref)
```

Comment:
I'm not in the habit of doing that. Maybe because I use a rainbow indent plugin in my editor (probably designed for the yaml crowd) I don't have a problem reading this stuff. It is more difficult in a PR.